src/ray/object_manager/plasma/store.h

```diff
@@ -206,9 +206,24 @@ class PlasmaStore {
   void DisconnectClient(const std::shared_ptr<Client> &client)
       ABSL_EXCLUSIVE_LOCKS_REQUIRED(mutex_);
 
-  Status ProcessMessage(const std::shared_ptr<Client> &client,
-                        plasma::flatbuf::MessageType type,
-                        const std::vector<uint8_t> &message) ABSL_LOCKS_EXCLUDED(mutex_);
+  /// Handle an unexpected connection error from the client.
+  /// The client will be disconnected and no more messages will be processed.
+  ///
+  /// \param client The client whose connection the error occurred on.
+  /// \param error The error details.
+  void HandleClientConnectionError(const std::shared_ptr<Client> &client,
+                                   const boost::system::error_code &error)
+      ABSL_LOCKS_EXCLUDED(mutex_);
+
+  /// Process a message from the client.
+  ///
+  /// \param client The client that the message is from.
+  /// \param type The message type.
+  /// \param message The message data.
+  Status ProcessClientMessage(const std::shared_ptr<Client> &client,
```

Comment:
@edoakes completely unrelated to this PR, do you know why do we use flatbuffer here?
I could try to avoid using it unless performance critical, one big downside is it's untyped thus no type checking; made a few mistakes in the past.

Reply:
Some of the worker<>raylet messages are on the critical path for worker task execution, so there is a performance consideration. That said, I'm not sure if there has been comprehensive benchmarking done on this, at least in a long time.

We can also mitigate the typing issue by doing a better job quarantining the code that casts the data to the relevant FB message.